### PR TITLE
Run github action with python 3.9

### PR DESCRIPTION
**Pull request by jenshnielsen**
This seems to be working fine. A few packages needed to be build from source but they are cached so the speed impact is minimal.
@astafan8

### CODE REVIEW

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -38,7 +40,7 @@ jobs:
         path: ${{ steps.prepare-cache-pip.outputs.cache-dir }}
         key: ${{ runner.os }}-${{ matrix.python-version }}-pip-${{ hashFiles('**/requirements.txt') }}-${{ hashFiles('**/docs_requirements.txt') }}-${{ hashFiles('**/setup.py') }}
         restore-keys: |
-          ${{ runner.os }}-pip-
+          ${{ runner.os }}-${{ matrix.python-version }}-pip-
```
Comment (astafan8) [inline .github/workflows/docs.yaml:43]: that's a good one!

File: .github/workflows/docs.yaml [hunk 2]
```diff
@@ -25,6 +25,8 @@ jobs:
       uses: ./.github/actions/setup-ubuntu-latest-xvfb
     - name: install pandoc ubuntu
       run: sudo apt install pandoc
+    - name: install libhdf5-dev ubuntu
+      run: sudo apt install libhdf5-dev
```
Comment (astafan8) [inline .github/workflows/docs.yaml:29]: so docs need only hdf5 while the tests need a few more packages for dealing with XML? interesting

**Review by astafan8 — APPROVED** [review]
python 3.9 - ready! :)

File: .github/workflows/docs.yaml (hunk 2 above)
Comment (jenshnielsen) [reply .github/workflows/docs.yaml:29]: Yes some of the awg tests uses lxml which needs these extra packages to be build from source

**Pull request merged**